std-bits/base/src/main/java/org/enso/base/ObjectComparator.java

```diff
@@ -102,7 +102,7 @@ public int compare(Object thisValue, Object thatValue) {
 
     // Booleans
     if (thisValue instanceof Boolean thisBool && thatValue instanceof Boolean thatBool) {
-      if (thisBool == thatBool) {
+      if (thisBool.booleanValue() == thatBool.booleanValue()) {
```

Comment:
This is an unrelated change but it seemed risky - I don't think we can assume that only 2 cached instances of boxed `Boolean` will ever appear in the program so if we want to use `==` we should unbox them first.

Reply:
> Cleaner to use `thisBool.equals(thatBool)`?

It is probably a micro-optimization, but technically `thisBool.equals(thatBool)` does more work:
```java
public boolean equals(Object obj) {
    if (obj instanceof Boolean) {
        return value == ((Boolean)obj).booleanValue();
    }
    return false;
}
```
and it boils down to comparing `booleanValue()`s anyway.

IMO 'unboxing' is OK here